Link fixes + WR ticket (#2206)
cloudflare/cloudflare-docs: products/fundamentals/src/content/internet/protocols/dns.md

## The DNS Namespace

> The hierarchy of DNS namespace is well defined: DNS names consist of labels that are separated by dots. Thus `www.example.com.` consists of 4 labels `www` (leaf) `example` (domain) `com` (TLD) and `.` i.e. the root. Resolvers search by starting at the longest match of the labels, i.e., if it only knows about the root then, it starts the search there. If they know about .com they start there.

Reviewer: ```suggestion
The hierarchy of DNS namespace is well defined: DNS names consist of labels that are separated by dots. Thus `www.example.com.` consists of 4 labels `www` (leaf) `example` (domain) `com` (TLD) and `.` (i.e., the root). Resolvers search by starting at the longest match of the labels, i.e., if it only knows about the root then, it starts the search there. If resolvers know about `.com`, they start there.
```

> There are thirteen rootservers maintained by 12 different organizations. The root zone is maintained by IANA, which is a part of ICANN, and is published by Verisign which operates two of the root servers. The contents of the root zone is a list of the authoritative servers for each TLD, but not much else. Thus, if a root DNS server receives a query for `www.example.com` it will answer with what is called a referral to `.com` resolvers. The answer from the root server contains a set of records called NS (Name Server). These records list the nameservers which should have more information about the requested zone, i.e., the “.com” servers.

Reviewer: ```suggestion
There are 13 root servers maintained by 12 different organizations. The root zone is maintained by the Internet Assigned Numbers Authority (IANA), which is a part of ICANN, and is published by Verisign which operates two of the root servers. The contents of the root zone is a list of the authoritative servers for each TLD, but not much else. Thus, if a root DNS server receives a query for `www.example.com` it will answer with what is called a referral to `.com` resolvers. The answer from the root server contains a set of records called NS (Name Server). These records list the nameservers which should have more information about the requested zone, i.e., the `.com` servers.
```

## DNSSEC

> The security extensions to DNS add protection for DNS records, and allow the resolvers and applications to authenticate the data received. These powerful additions will mean that all answers from DNS can be trusted. Earlier we described how DNS resolution starts at the root nameserver, and works down (e.g. for `www.example.com` it starts at the root server, then “com”, then “example.com”). This is the same way that trust is conferred via DNSSEC. The DNS root is the definitive root of trust, and a chain of trust is built to the root from any DNS entry. This is a lot like the chain of trust used to validate TLS/SSL certificates, except that, rather than many trusted root certificates, there is one trusted root key managed by the DNS root maintainer IANA.

Reviewer: ```suggestion
The security extensions to DNS add protection for DNS records, and allow the resolvers and applications to authenticate the data received. These powerful additions will mean that all answers from the DNS can be trusted. Earlier you learned how DNS resolution starts at the root nameserver, and works down (e.g., for `www.example.com` it starts at the root server, then “com”, then “example.com”). This is the same way that trust is conferred via DNSSEC. The DNS root is the definitive root of trust, and a chain of trust is built to the root from any DNS entry. This is a lot like the chain of trust used to validate TLS/SSL certificates, except that, rather than many trusted root certificates, there is one trusted root key managed by the DNS root maintainer IANA.
```
